Release/include/cpprest/details/cpprest_compat.h

```diff
@@ -71,14 +71,23 @@
 
 #ifdef _NO_ASYNCRTIMP
 #define _ASYNCRTIMP
+#define _ASYNCRTIMP_TYPEINFO
 #else // ^^^ _NO_ASYNCRTIMP ^^^ // vvv !_NO_ASYNCRTIMP vvv
 #ifdef _ASYNCRT_EXPORT
 #define _ASYNCRTIMP __declspec(dllexport)
 #else // ^^^ _ASYNCRT_EXPORT ^^^ // vvv !_ASYNCRT_EXPORT vvv
 #define _ASYNCRTIMP __declspec(dllimport)
 #endif // _ASYNCRT_EXPORT
+
+#ifdef __clang__
+#define _ASYNCRTIMP_TYPEINFO __attribute__((visibility("default")))
+#else // ^^^ __clang__ ^^^ // vvv !__clang__ vvv
+#define _ASYNCRTIMP_TYPEINFO
+#endif // __clang__
```

Comment:
I don't think clang is the correct thing to use here, it should be
"if _NO_ASYNCRTIMP is not defined and we're not building for windows"

Reply:
I was thinking about that, but since the clang compiler can also be used on Windows, I thought __clang__ would be better for cross compiler compatibility.

Reply:
ah, but the thing is you actually don't want to export the "class" when clang is running in msvc ABI mode (in this case it will define `__clang__` and `_WIN32`.